tests/unit/services/apigateway/test_uri_interpolation.py

```diff
@@ -0,0 +1,128 @@
+                "https://example.com/foo?q=stageValue",
+            ),
+            (
+                # AWS URI action or path components
+                "arn:aws:apigateway:<region>:<service>:${stageVariables.stageVar}",
+                "arn:aws:apigateway:<region>:<service>:stageValue",
+            ),
+            (
+                # AWS integration Lambda function name
+                "arn:aws:apigateway:<region>:lambda:path/2015-03-31/functions/arn:aws:lambda:<region>:<account_id>:function:${stageVariables.stageVar}/invocations",
+                "arn:aws:apigateway:<region>:lambda:path/2015-03-31/functions/arn:aws:lambda:<region>:<account_id>:function:stageValue/invocations",
+            ),
+            (
+                # AWS integration Lambda function version/alias
+                "arn:aws:apigateway:<region>:lambda:path/2015-03-31/functions/arn:aws:lambda:<region>:<account_id>:function:<function_name>:${stageVariables.stageVar}/invocations",
+                "arn:aws:apigateway:<region>:lambda:path/2015-03-31/functions/arn:aws:lambda:<region>:<account_id>:function:<function_name>:stageValue/invocations",
+            ),
+            (
+                # Amazon Cognito user pool for a COGNITO_USER_POOLS authorizer.
+                "arn:aws:cognito-idp:<region>:<account_id>:userpool/${stageVariables.stageVar}",
+                "arn:aws:cognito-idp:<region>:<account_id>:userpool/stageValue",
+            ),
+            (
+                # AWS user/role integration credentials ARN
+                "arn:aws:iam::<account_id>:${stageVariables.stageVar}",
+                "arn:aws:iam::<account_id>:stageValue",
+            ),
+        ],
+    )
+    def test_uri_stage_variables_interpolation(self, uri, expected):
+        # test values taken from the documentation
+        # https://docs.aws.amazon.com/apigateway/latest/developerguide/aws-api-gateway-stage-variables-reference.html#stage-variables-in-integration-HTTP-uris
+        stage_variables = {
+            "stageVar": "stageValue",
+            "stageDomain": "example.com",
+            "moreVar": "var",
+        }
+        rendered = render_uri_with_stage_variables(uri=uri, stage_variables=stage_variables)
+        assert rendered == expected
+
+    def test_uri_interpolating_with_only_curly_braces(self):
+        uri_with_only_braces = "https://test.domain.example/root/{path}"
+        stage_variables = {"path": "value"}
+        rendered = render_uri_with_stage_variables(
+            uri=uri_with_only_braces, stage_variables=stage_variables
+        )
+        assert "{path}" in rendered
+
+    def test_uri_interpolating_with_no_variable(self):
+        uri_with_only_braces = "https://test.domain.example/root/${path}"
+        stage_variables = {}
+        rendered = render_uri_with_stage_variables(
+            uri=uri_with_only_braces, stage_variables=stage_variables
+        )
+        assert rendered == "https://test.domain.example/root/"
```

Comment:
Nice I was wondering if that was the result we would get! :+1: 
I assume the same happens if you use `${stageVariables.path}`?

Reply:
Ah! Good catch! This is actually wrong, this should be the result when using `${stageVariables.path}`, I believe if we were to use `${path}` it might just try to interpolate it as a path param, so if no path param named `path` it will leave it to `${path}`, otherwise might render `$<path-param>`, I need to try it... oops 😬